### ISSUE TITLE
[PR Proposal - AnimationBuilder] Add styles when animation completes

### ISSUE THREAD

**Issue by ericmartinezr** [angular/angular#4794]
**TL;DR** Basically I want to be able to replace some styles added during the animation when it completes

```
FromStyles ---> ToStyles ----> Animation Completes ---> Add new styles/Replace existing ones
```

**Long version**

I want to solve the problem with animating from `height:0` to `height:auto`, which is not possible. 
How am I doing it so far? I animate from `height:0` to `height:XX` (fixed size) and after that I replace `height:XX` with `height:auto`

I can do this currently with this piece of code (from memory)

**Current working code** ([plnkr repro](http://plnkr.co/edit/op3Kzd?p=preview))
```
var animationCss = AnimationBuilder.css();
animationCss
	.setStyles({'height':'0'}, {'height':'100px'})
	.start(element)
	.onComplete(() => 
		animationCss.setToStyles({'height':'auto'}).start(element);
	);
```
The problem with the code above is 

- It's too verbose for just trying to do a few things in one animation
- I'm actually creating two different animations (each `.start()` returns a new animation)

So I came up with this solution that I would like to have your feedback for (maybe there's a better option that I'm not aware of)

**Proposal** ([plnkr repro](http://plnkr.co/edit/JxWEap?p=preview) with custom ng2 building)
*You can see my proposal here https://github.com/angular/angular/compare/master...ericmartinezr:animation-test*
```
var animationCss = AnimationBuilder.css();
animationCss
	.setStyles({'height':'0'}, {'height':'100px'})
	// When the animation completes I replace `height:100px` with `height:auto`
	.setOnCompleteStyles({'height':'auto'})
	.start(element);
```
Does the same in one animation and in 2~3 lines less than the current way. Clearly the most important point is the fact of using only one animation.

Note that these styles wouldn't be part of the animation itself, they just would be added when everything is completed. 
This is for animations where the content may be dynamic so fixed sizes wouldn't work (comment the lines where I add `height:auto` in both cases and you'll see that with fixed size you can't see the dynamic div).

I don't know if this is too specific for my case only, so I opened this issue instead of sending a PR to know what you think about it.

**Comment by vicb** [comment]
You should check the live stream for [Angular Connect](http://angularconnect.com/) next week. @matsko and @robertmesserle will have a talk on animation, you'll probably get some answers.

**Comment by matsko** [comment]
@ericmartinezr this implementation of animation is just a temporary system. The real system will be announced next week.

**Comment by ericmartinezr** [comment]
Ohhh... ok! Thanks for the heads up, I'll try to achieve this with the real system (if you didn't do it already :P). Closing.
